xmake/rules/c++/modules/modules_support/msvc.lua

```diff
@@ -341,60 +371,98 @@ function build_modules_for_batchjobs(target, batchjobs, objectfiles, modules, op
+                                table.join2(flags, {flags_[i], flags_[i + 1]})
+                            end
                         end
-                        target:data_set("cxx.add_modules_mapflags", true)
                     end
-                    local args = {"-c", "-Fo" .. objectfile, interfaceflag, ifcoutputflag, bmifile, provide.sourcefile}
-                    os.vrunv(compinst:program(), table.join(compinst:compflags({target = target}), common_args, args), {envs = vcvars})
-                end, {dependfile = target:dependfile(bmifile), files = {provide.sourcefile}})
-            end)
-            if m.requires then
-                moduleinfo.deps = table.keys(m.requires)
+
+                    depend.on_changed(function()
+                        progress.show((index * 100) / total, "${color.build.object}generating.cxx.module.bmi %s", name)
+                        local objectdir = path.directory(objectfile)
+                        if not os.isdir(objectdir) then
+                            os.mkdir(objectdir)
+                        end
+                        local args = {"-c", "-Fo" .. objectfile, interfaceflag, ifcoutputflag, bmifile, provide.sourcefile}
+                        os.vrunv(compinst:program(), table.join(compinst:compflags({target = target}), common_args, flags or {}, args), {envs = vcvars})
+                    end, {dependfile = target:dependfile(bmifile), files = {provide.sourcefile}})
+                    _add_module_to_mapper(target, referenceflag, name, bmifile, flags)
+                end)
+                if module.requires then
+                    moduleinfo.deps = table.keys(module.requires)
+                end
+                moduleinfo.name = name
+                modulesjobs[name] = moduleinfo
+                target:add("objectfiles", objectfile)
+            else
+                if module.requires then
+                    local moduleinfo = {}
+                    moduleinfo.name = module.cppfile
+                    moduleinfo.deps = table.keys(module.requires)
+                    moduleinfo.sourcefile = module.cppfile
+                    moduleinfo.job = batchjobs:newjob(module.cppfile, function(index, total) 
+                        function contains(t, v)
+                            for _, flag in pairs(t) do
+                                if table.contains(flag, v) then
+                                    return true
+                                end
+                            end
+                            return false
+                        end
+                        -- append module mapper flags
+                        -- @note we add it at the end to ensure that the full modulemap are already stored in the mapper
+                        local flags_ = get_requiresflags(target, module.requires)
+                        local flags = {}
+                        for i=1, #flags_, 2 do
```

Comment:
`i = 2`

Reply:
why =2 ? i need to begin at one to get the argument too

Reply:
my mistake, `i = 1` just modify space